test/src/commonMain/kotlin/test/link/Parent.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package test.link
+
+import io.realm.runtimeapi.RealmModel
+import io.realm.runtimeapi.RealmObject
+
+@RealmObject
+class Parent : RealmModel {
+    var child: Child? = null
```

Comment:
What happens if we define a default value `child = Child("foo)` do you think we can prevent such behaviour now or when we implement default values?

Reply:
I guess we could just `error` from the compiler plugin!?

Reply:
Will postpone this till default value implementation in #106